src/ids-tooltip/ids-tooltip.js

```diff
@@ -0,0 +1,377 @@
+        this.visible = false;
+      });
+
+      // Long Press
+      this.onEvent('longpress.tooltip', targetElem, () => {
+        this.visible = true;
+      }, { delay: 1000 });
+
+      // Keyboard Focus event
+      this.onEvent('keyboardfocus.tooltip', targetElem, () => {
+        this.visible = true;
+      });
+
+      /* istanbul ignore next */
+      this.onEvent('focusout.tooltip', targetElem, () => {
+        this.visible = false;
+      });
+
+      /* istanbul ignore next */
+      this.onEvent('click.popup', this.popup, () => {
+        this.visible = false;
+      });
+    }
+
+    // Events to show on click
+    if (this.trigger === 'click') {
+      this.onEvent('click.tooltip', targetElem, (e) => {
+        this.popup.alignTarget = e.currentTarget;
+        if (this.visible) {
+          this.visible = false;
+          return;
+        }
+        this.visible = true;
+      });
+
+      this.onEvent('click.popup', this.popup, () => {
+        this.visible = false;
+      });
+    }
+
+    // Events to show on focus
+    if (this.trigger === 'focus') {
+      this.onEvent('focusin.tooltip', targetElem, (e) => {
+        this.popup.alignTarget = e.currentTarget;
+        this.visible = true;
+      });
+
+      this.onEvent('focusout.tooltip', targetElem, () => {
+        this.visible = false;
+      });
+    }
+  }
+
+  /**
+   * Setup the popup
+   * @private
+   */
+  #configurePopup() {
+    // Popup settings / config
+    this.popup.type = 'tooltip';
+    this.popup.align = `${this.placement}, center`;
+    this.popup.arrow = this.placement;
+
+    if (this.placement === 'top' || this.placement === 'bottom') {
+      this.popup.x = 0;
+      this.popup.y = 12;
+    }
+    if (this.placement === 'left' || this.placement === 'right') {
+      this.popup.x = 12;
+      this.popup.y = 0;
+    }
+  }
+
+  /**
+   * Update the aria attributes with the correct contents
+   * @private
+   */
+  #updateAria() {
+    // For ellipsis based tooltips we dont do this
+    if (this.state.noAria) {
+      return;
+    }
+    this.popup.alignTarget = typeof this.target === 'object'
+      ? this.target
+      : document.querySelectorAll(this.target)[0];
+
+    const id = `${this.id || 'ids'}-tooltip`;
+    if (this.popup.alignTarget && this.popup.alignTarget.querySelector(`#${id}`)) {
+      this.popup.alignTarget.querySelector(`#${id}`).textContent = this.textContent;
+      return;
+    }
+
+    if (this.popup.alignTarget) {
+      const ariaSpan = `<ids-text id="${id}" audible="true">${this.textContent}</ids-text>`;
+      this.popup.alignTarget.insertAdjacentHTML('beforeend', ariaSpan);
+      this.popup.alignTarget.setAttribute('aria-describedby', `#${id}`);
+    }
+  }
+
+  /**
+   * Show the tooltip (use visible for public API)
+   * @private
+   */
+  async #show() {
+    // Trigger a veto-able `beforeshow` event.
+    let canShow = true;
+    const beforeShowResponse = (/** @type {any} */ veto) => {
```

Comment:
```suggestion
    const beforeShowResponse = (veto) => {
```
Do we still need these after the TS changes?